ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -222,12 +222,12 @@ private void initialize(final DatagramSocket socket, final int tcpPort) {
   }
 
   /**
-   * This is the exception handler for uncontrolled exceptions ocurring in the packet handlers.
+   * This is the exception handler for uncontrolled exceptions occurring in the packet handlers.
    *
-   * @param throwable the exception that was raised
+   * @param exception the exception that was raised
    */
-  private void handleException(final Throwable throwable) {
-    System.out.println(throwable);
+  private void handleException(final Throwable exception) {
+    System.err.println(String.format("Packet handler exception: %s.", exception.getMessage()));
```

Comment:
Good spot, but this should be sent to a logger, not `System.err` or `System.out` and it should include the stack trace, so:
`LOG.error("Packet handler exception", exception);`

We may want to check the exception type though - `IOException` is fairly expected in this area so probably should be debug level.